lib/server.js

```diff
@@ -48,6 +59,7 @@ function createServer (options, httpHandler) {
     } else {
       listenOptions.cb = cb
     }
+    addAbortSignalOption(this.close, server, listenOptions)
```

Comment:
You dont need a new function. You can do everything here. 

```suggestion
    
    if (listenOptions.signal) {
      const onAbortedCallback = () => { this.close() }
      listenOptions.signal.addEventListener('abort', onAbortedCallback)
      server.once('close', () => listenOptions.signal.removeEventListener('abort', onAbortedCallback))
    }
```

Reply:
Don't you prefer short functions?

Reply:
Generally yes, but I dont see a benefit here. Just looks like it is ripped out of the place it should have been in the first place.